File: mod/ondb/lib/js/query.js
import { Ondb } from './ondb_core.js';

let etype = '';
for (let i = 2; i < Bun.argv.length; i++) {
  const a = Bun.argv[i];
  if (a.startsWith('--type=')) etype = a.slice(7);
}

const db = new Ondb();
const input = await Bun.stdin.text();
db.load(input);

const out = [];
for (let i = 0; i < db._entity_order.length; i++) {
  const eid = db._entity_order[i];
  const e = db.entities.get(eid);
  if (!e || !e.alive) continue;
  if (etype && e.type !== etype) continue;
  out.push(`${eid}\t${e.type}\t${e.props.name || ''}`);
}
process.stdout.write(out.join('\n') + '\n');

File: mod/ondb/lib/py/query.py
import sys, os
sys.path.insert(0, os.path.dirname(__file__))
from ondb_core import Ondb

etype = ''
for a in sys.argv[1:]:
    if a.startswith('--type='): etype = a.split('=', 1)[1]

db = Ondb()
db.load(sys.stdin)
for eid, typ, name in db.query(etype or None):
    print(f"{eid}\t{typ}\t{name}")
